web/packages/teleterm/src/ui/Vnet/VnetConnectionItem.tsx

```diff
@@ -0,0 +1,235 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Text, ButtonIcon, Flex, rotate360 } from 'design';
+import * as icons from 'design/Icon';
+import { copyToClipboard } from 'design/utils/copyToClipboard';
+
+import { ConnectionStatusIndicator } from 'teleterm/ui/TopBar/Connections/ConnectionsFilterableList/ConnectionStatusIndicator';
+import { ListItem } from 'teleterm/ui/components/ListItem';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+
+import { useVnetContext } from './vnetContext';
+
+/**
+ * VnetConnection is the VNet entry in Connections. Also used as the topmost item in VnetSliderStep.
+ */
+export const VnetConnectionItem = (props: {
+  onClick: () => void;
+  title: string;
+  showBackButton?: boolean;
+  showHelpButton?: boolean;
+}) => {
+  const { status, start, stop, startAttempt, stopAttempt } = useVnetContext();
+  const indicatorStatus =
+    startAttempt.status === 'error' || stopAttempt.status === 'error'
+      ? 'error'
+      : status === 'running'
+        ? 'on'
+        : 'off';
+
+  return (
+    <ListItem
```

Comment:
I don't know if this is easy to do, but I'm not sure if the entire VNet header should be clickable if is used as the item in `VnetSliderStep`. Instead, I would make clickable only the "back arrow". 
Now it suggests that you would be taken to some other VNet screen (because the entire section is hovered), while it just goes back to all connections.

But maybe it doesn't matter since you want to change the UI and make the panel expandable.

Reply:
I'd rather have a bigger click target than a single icon (see also https://github.com/gravitational/teleport/issues/37921#issuecomment-2025281399). But yeah, with an expanding animation this might not even be the case. Maybe the arrow should somehow get hover styles if I hover on the block.